app/elements/code-editor/code-editor.html

```diff
@@ -66,26 +61,40 @@
 				this.$.editor.getSession().setValue(text);
 			},
 			setMode: function(lang){
-				var mode = 'ace/mode/java', fileName = 'Application.java';
+				var mode = 'ace/mode/java';
 				switch (lang) {
 					case 'py':
 						mode = 'ace/mode/python';
-						fileName = 'app.py';
 						break;
 					case 'c':
 						mode = 'ace/mode/c_cpp';
-						fileName = 'app.c';
 						break;
 					case 'cpp':
 						mode = 'ace/mode/c_cpp';
-						fileName = 'app.cpp';
+						break;
+					case 'robot':
+						mode = 'ace/mode/c_cpp';
 						break;
 				}
-				this.fileName = fileName;
 				this.$.editor.getSession().setMode({
 					path: mode,
 					v: Date.now()
 				});
+			},
+			_getFile: function(lang){
```

Comment:
Ideally we should not need this, and rather use the filename we get from the templates.